src/rebar_prv_common_test.erl

```diff
@@ -215,6 +216,13 @@ add_hooks(Opts, State) ->
 select_tests(_, _, {error, _} = Error, _) -> Error;
 select_tests(_, _, _, {error, _} = Error) -> Error;
 select_tests(State, ProjectApps, CmdOpts, CfgOpts) ->
+    %% set application env if sys_config argument is provided
+    SysConfigs = sys_config_list(CmdOpts, CfgOpts),
+    Configs = lists:flatmap(fun(Filename) ->
+                                rebar_file_utils:consult_config(State, Filename)
+                            end, SysConfigs),
+    rebar_utils:reread_config(Configs),
```

Comment:
Configurations won't hold properly if the target application hasn't been loaded. Before running this, calling `application:load(App)` on each apps in the config may be required, sadly.